src/common/SteelSheets.hpp

```diff
@@ -0,0 +1,122 @@
+#pragma once
+#include "marlin_client.h"
+#include "eeprom.h"
+
+class SteelSheets {
+public:
+    static constexpr float Z_OFFSET_MIN = -2.0F;
+    static constexpr float Z_OFFSET_MAX = 0.0F;
+
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief Sets offset for sheet
+    ///
+    /// @param[in] index Index of the sheet profile
+    /// @param[in] offset of the sheet
+    /// @return True when successful, false if not
+    static bool SetSheetOffset(uint32_t index, float offset);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief Iterate across the profiles and switch to the next calibrated.
+    ///
+    /// Printer use print sheet profile on the index 0 as a default so the method
+    /// in the worst case iterate across entire profiles and return index 0 when
+    /// not any other profile is calibrated yet.
+    /// @return Index of the next calibrated profile.
+    static uint32_t NextSheet();
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief Determine if the given sheet profile is calibrated.
+    ///
+    /// In case the index of the given print sheet profile is bigger than the
+    /// MAX_SHEETS method return false.
```

Comment:
> In case the index of the given print sheet profile is bigger than the MAX_SHEETS method return false.
To my opinion either documentation or implementation is wrong. If the index is out of range
``` c++
Sheet { "UNDEF", 0 };
```
is returned by the `SteelSheets::getSheet(out_of_range_index)` And 0 is not nearly equal to FLT_MAX, so such "UNDEF" sheet is assumed to be calibrated.